command/scripting.go

```diff
@@ -0,0 +1,370 @@
+package command
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/rest/client"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/google/shlex"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/send"
+	"github.com/mongodb/jasper/options"
+	"github.com/mongodb/jasper/scripting"
+	"github.com/pkg/errors"
+)
+
+type scriptingExec struct {
+	// Harness declares the implementation of the scripting
+	// harness to use to execute this code.
+	Harness string `mapstructure:"harness"`
+
+	////////////////////////////////
+	//
+	// Exeuction Options
+
+	// Specify the command to run as a string that Evergreen will
+	// split into an argument array.
+	Command string `mapstructure:"command"`
+	// Specify the command to run as a list of arguments.
+	Args []string `mapstructure:"args"`
+	// Specify the content of a script to execute in the
+	// environment.
+	Script string `mapstructure:"script"`
+	// Specify a list of directories to add to the PATH of the
+	// environment.
+	Path []string `mapstructure:"add_to_path"`
+	// Specific environment variables to be added to the
+	// environment of the scripting commands executed
+	Env map[string]string `mapstructure:"env"`
+
+	////////////////////////////////
+	//
+	// Harness Options
+
+	// CacheDurationSeconds describes the total number of seconds
+	// that an environment should be stored for.
+	CacheDurationSeconds int `mapstructure:"cache_duration_secs"`
+	// CleanupHarness forces the command to cleanup the harness
+	// after the command returns. When this is false, the harness
+	// will persist between commands. These harnesses are within
+	// the working directory and so cleaned up between tasks or
+	// task groups regardless.
+	CleanupHarness bool `mapstructure:"cleanup_harness"`
+	// LockFile describes the path to the dependency file
+	// (e.g. requirements.txt if it exists,) that lists your
+	// dependencies. Not all environments support Lockfiles
+	LockFile string `mapstructure:"lock_file"`
+	// Packages are a list of dependencies that will be installed
+	// in your environment.
+	Packages []string `mapstructure:"packages"`
+	// HarnessPath should be the path to your local environment
+	// (e.g. GOPATH or VirtualEnv.) Specify a subpath of the
+	// working directory.
+	HarnessPath string `mapstructure:"harness_path"`
+	// HostPath is the path to the hosting interpreter or binary,
+	// where appropriate. This should be the path to the python
+	// interpreter or go binary.
+	HostPath string `mapstructure:"host"`
```

Comment:
Can we name this something other than host path? I have no clue what host path means. Also this documentation seems wrong - this will set GOROOT, not the go binary path (which is hard-coded into Jasper as `GOROOT/bin/path`).